lib/metadata/bucketfile/backend.js

```diff
@@ -429,31 +260,38 @@ class BucketFileInterface {
                 return cb(err);
             }
             let cbDone = false;
-            const stream = db.createReadStream(requestParams);
-            stream
-                .on('data', e => {
-                    if (extension.filter(e) < 0) {
-                        stream.emit('end');
-                        stream.destroy();
-                    }
-                })
-                .on('error', err => {
-                    if (!cbDone) {
-                        this.unRef();
-                        cbDone = true;
-                        log.error('error listing objects',
-                                  { error: err });
-                        cb(errors.InternalError);
-                    }
-                })
-                .on('end', () => {
-                    if (!cbDone) {
-                        this.unRef();
-                        cbDone = true;
-                        const data = extension.result();
-                        cb(null, data);
-                    }
-                });
+            db.createReadStream(requestParams, (err, stream) => {
```

Comment:
I think it's more standard here to use `createReadStream` as a synchronous call since we catch the events `error` of the result stream.